### DISCUSSION TITLE
Is this possible to include libraries in .langium file?

### DISCUSSION THREAD

**Discussion by idiosyncrasy00** [eclipse-langium/langium#977] [Q&A] [answered]
Hi, I already have a Xtext file with grammar rule which imports an Ecore and several libraries from Eclipse. 
I wonder if langium enables the feature of using external libraries to define your grammar besides its internal ones.
Thanks.
P/s: I've read a similar discussion about this, [https://github.com/langium/langium/discussions/397], apparently it's not gonna be a thing anytime soon, but still, a year later and hopefully there is a way to resolve this.

**Comment by msujew** [comment, marked as answer]
Hey @LockedAwya,

so for one, importing other grammars can be done through a simple import mechanism. See [here for an example](https://github.com/langium/langium/blob/main/examples/requirements/src/language-server/tests.langium).

We have also added a type declaration format similar to Ecore [back in v0.3](https://github.com/langium/langium/pull/406). However, it is not as extensive as Ecore/EMF is. This is by design to keep the library simple.

The two features above can obviously be combined. So you can build a `.langium` file purely consisting of interfaces, while another implements the grammar rules for these elements.

By the way, we have also released a tool to automatically migrate from Xtext grammars to Langium. It might not do a full migration for complex setups though: https://github.com/TypeFox/xtext2langium
